File: pow.js
export const pow = (blockdata, difficulty) => {
	let nounce = 'NOT-FOUND',
		startTime = Data.now(),
		endTime = start,
		tryNounce = 0,
		sha = null;

	while (tryNounce < 1000000000) {
		let newData = blockdata + tryNounce,
			sha = new sha256(newData);
		if (sha.startWith(difficulty)) {
			end = Date.now();
			nounce = tryNounce;
			console.log(sha.digest());
			break;
		}
		tryNounce++;
	}
	let timetaken = end - start;
	if (timetaken === 0) {
		console.log(`Difficulty=${difficulty}, Nounce not found !!!`);
	} else {
		console.log(`Difficulty=${difficulty}, Time(ms)=${timetaken}, Nounce=${nounce}, Hash=${sha.digest()}`);
	}
};

File: pow.py

from datetime import datetime as dt
import hashlib


def proof_of_work(bloockdata, difficulty):
    nounce = 'NOT-FOUND'
    startTime = dt.now()
    endTime = startTime
    tryNounce = 0

    while tryNounce < 10000000:
        newData = bloockdata + str(tryNounce)
        sha = hashlib.sha256()
        sha.update(newData.encode('utf-8'))
        if sha.hexdigest().startswith(difficulty):
            endTime = dt.now()
            nounce = tryNounce
            break
        tryNounce+=1

    timetaken = endTime - startTime
    if timetaken == 0:
        print ('Difficulty=', difficulty, ' Nounce not found!!!!')
    else:
        print (
            'Difficulty =',
            difficulty,
            'Time =',
            timetaken,
            'Nounce =',
            nounce,
            'Hash =',
            sha.hexdigest(),
            )



			